byway/src/main/java/com/google/sps/servlets/PlacesServlet.java

```diff
@@ -52,4 +73,31 @@ private String formatLocation(PlaceType location) {
     place = place.replace('_', ' ');
     return place;
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String tripIdString = request.getParameter("tripId");
+    Long tripId = Long.parseLong(tripIdString);
+    Key specificTripKey = KeyFactory.createKey("trip", tripId);
+    ArrayList<String> interests = findInterests(request);
+    Entity trip;
+    try {
+      trip = datastore.get(specificTripKey);
+      trip.setProperty("interests", interests);
+      datastore.put(trip);
+    } catch(EntityNotFoundException e) {
+      //thrown automatically
+    }
+  }
+
+  /**
+   * Converts the JSON of interests selected and parses it into an ArrayList of Strings.
+   * @param  request servlet request
+   * @return list of interests
+   */
+  private ArrayList<String> findInterests(HttpServletRequest request) {
+    String interestsAsJSON = request.getParameter("interests");
```

Comment:
If you want to load something from the body of a post request rather than using query parameters, you can use `gson.fromJson(request.getReader(), ARRAYLIST_STRING)`. When doing this, the client would put the json string in the `body` of the post request, rather than adding it as query parameters.

Reply:
Fixed